library/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/listener/WindowsOnDrawListener.kt

```diff
@@ -50,8 +51,10 @@ internal class WindowsOnDrawListener(
             }
 
         if (nodes.isNotEmpty()) {
-            blockingQueueItem.nodes = nodes
-            blockingQueueHandler.update()
+            item.nodes = nodes
+            if (item.isReady()) {
```

Comment:
also why do you check here if `item.isReady()` if we already check it when we loop through the queue ?